File: SharedprefenceDemo/app/src/main/java/com/example/sharedprefencedemo/MainActivity.java
package com.example.sharedprefencedemo;

import androidx.appcompat.app.AppCompatActivity;

import android.content.SharedPreferences;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity {

TextView textView;
EditText editText;
Button button;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        textView=findViewById(R.id.textView);
        editText=findViewById(R.id.editText);
        button=findViewById(R.id.button2);
        SharedPreferences sharedPreferences=getSharedPreferences("mydata",MODE_PRIVATE);

        button.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                SharedPreferences.Editor editor=sharedPreferences.edit();
                String str= editText.getText().toString();
                 editor.putString("tata",str);
                 editor.apply();

            }
        });
        String str2=  sharedPreferences.getString("tata","default");
        textView.setText(str2);

    }
}

File: ContextMenu/app/src/main/java/com/example/contextmenu/MainActivity.kt
package com.example.contextmenu

import android.content.Context
import android.content.SharedPreferences
import android.content.SharedPreferences.Editor
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import android.view.ContextMenu
import android.view.MenuItem
import android.view.View
import android.widget.TextView
import android.widget.Toast

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        val textView=findViewById<TextView>(R.id.textView)
        registerForContextMenu(textView)
 
    }

    override fun onCreateContextMenu(
        menu: ContextMenu?, v: View?, menuInfo: ContextMenu.ContextMenuInfo?) {
        menuInflater.inflate(R.menu.context_menu , menu)
    }

    override fun onContextItemSelected(item: MenuItem): Boolean {
        val itemId =item.itemId
        if (itemId==R.id.share)
            Toast.makeText(applicationContext,"Share To",Toast.LENGTH_SHORT).show()
        else if (itemId==R.id.forword)
            Toast.makeText(applicationContext,"Forword To",Toast.LENGTH_SHORT).show()
        else if (itemId==R.id.setting)
            Toast.makeText(applicationContext, "Setting", Toast.LENGTH_SHORT).show()
        return super.onContextItemSelected(item)
    }
}




